tfrecorder/beam_pipeline.py

```diff
@@ -311,12 +323,12 @@ def build_pipeline(
         | 'DiscardDataWriter' >> beam.io.WriteToText(
```

Comment:
Maybe it's good to write any discarded data even if `split_key` is not present. This would require partitioning the data differently though. I think easiest way is to partition the discarded data before you try to partition between train/val/test if split_key is present.

Reply:
I think you implemented this in a separate PR that we just merged in, so I believe it's addressed.